controllers/nodefeaturediscovery_controller.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2020-2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/cri-api/pkg/errors"
+	"k8s.io/klog"
```

Comment:
time for `k8s.io/klog/v2`?

Reply:
same comment as the CRD, this patch is big enough like to introduce more changes, I am planning on enhancing logging and adding changes to the CRD but not in this patch that is big enough.
Thanks for the reviews Mythi :)